.github/workflows/release.yml

```diff
@@ -0,0 +1,51 @@
+name: Release Chart
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  release:
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure Git
+        run: |
+          git config user.name "$GITHUB_ACTOR"
+          git config user.email "$GITHUB_ACTOR@users.noreply.github.com"
+
+      - name: Generate Chart
+        run: make helm-chart-generate
+
+      - name: Check for changes to generated helm chart
+        id: check_chart_changes
+        run: >-
+          git diff --quiet charts/
+          && echo "chart_changes=true" >> $GITHUB_OUTPUT
+          || echo "chart_changes=false" >> $GITHUB_OUTPUT
+
+      # If we generated a new helm chart, we need to commit it
+      # We need to do this step because our helm chart is not the source of truth, it's generated from starter
+      # Note that failing to do this doesn't break this release, but it can break future ones and make the release state confusing
+      - name: Commit generated chart
+        if: steps.check_chart_changes.outputs.chart_changes == 'true'
+        run: |
+          git pull -r origin/main
```

Comment:
```suggestion
          git pull --rebase origin/main
```

Reply:
@miles-w-3 FYI I'll commit this directly then approve but think we should use the long version of the option so it's more obvious.